searx/engines/dailymotion.py

```diff
@@ -24,21 +25,39 @@
 
 # search-url
 # see http://www.dailymotion.com/doc/api/obj-video.html
-search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'  # noqa
-supported_languages_url = 'https://api.dailymotion.com/languages'
+search_url = 'https://api.dailymotion.com/videos?fields=created_time,title,description,duration,url,thumbnail_360_url,id&sort=relevance&limit=5&page={pageno}&{query}'
+
+# The request query filters by 'localization', therefore instead of fetching
+# languages we need to fetch locales.
+supported_languages_url = 'https://api.dailymotion.com/locales'
 
 
-# do search-request
 def request(query, params):
-    if params['language'] == 'all':
-        locale = 'en-US'
-    else:
-        locale = match_language(params['language'], supported_languages)
 
-    params['url'] = search_url.format(
-        query=urlencode({'search': query, 'localization': locale}), pageno=params['pageno']
-    )
+    language = params['language']
+    if language == 'all':
+        language = 'en-US'
+
+    locale = babel.Locale.parse(language, sep='-')
+
+    # localization:
+    #   Change the default localization of the user.  This will affect results
+    #   language and content selection.
+    #   https://developers.dailymotion.com/api/#global-parameters
 
+    localization = locale.language
+    if locale.territory:
+        localization += '_' + locale.territory
+    if localization in supported_languages:
+        params['url'] = search_url.format(
+            query=urlencode({'search': query, 'localization': localization }),
+            pageno=params['pageno']
+        )
+    else:
+        params['url'] = search_url.format(
+            query=urlencode({'search': query}),
+            pageno=params['pageno']
+        )
```

Comment:
> * remains `ca` with this line --> I don't know how dailymotion deals with this unexpected value.

fixed

> Do you intend to send the language & region unfiltered to the engines?
At first sight, I think it is regression (`match_language` is going to map `ca-ES` to `ca` or `en`, so there is no unexpected value ; same `de-DE` is going to be mapped to `de` or `en`).

That is the issue of using this function here, the engine expect only one of the values from `supported_languages` .. here, dailymotion expect a locale from 

https://github.com/searxng/searxng/blob/74cf47fa1422384eb4e6fb047a9ff41aeeed5e5f/searx/data/engines_languages.json#L430-L481 

.. there is no language 'ca', 'en' .. available and it is wrong to send a language in the query.

Reply:
So in the case of `ca` (*), what should be done?
* should the engine skip the language parameter? (I'm not sure this is a good idea since the result are going to be related to the instance location).
* should the engine use `en-US`? or even `es_ES`? (`all` is mapped to `en-US`)
* should the engine returns zero result?

(*) currently [`ca-ES`](https://github.com/searxng/searxng/blob/5e7f46991104a6583676c6c9319a81b6f9a4b151/searx/languages.py#L9) but it is going to be `ca` [as far I understand](https://github.com/searxng/searxng/pull/1051#issuecomment-1093996357).